sale_order_whole_delivered_invoiceability/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    whole_delivered_invoiceability = fields.Boolean(
+        compute="_compute_whole_delivered_invoiceability",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("partner_id")
+    def _compute_whole_delivered_invoiceability(self):
+        for record in self:
+            if record.partner_id.whole_delivered_invoiceability:
+                record.whole_delivered_invoiceability = True
+
+    @api.depends("whole_delivered_invoiceability")
+    def _get_invoice_status(self):
+        # Intercept the invoice_status computed method to
+        # set it as not invoiceable if the delivered quantity
+        # is less than the ordered quantity.
+        res = super()._get_invoice_status()
+        for order in self:
+            if order.whole_delivered_invoiceability:
+                lines_to_invoice = order.order_line.filtered(
```

Comment:
This name is a little bit missleading, as we want to set lines that aren't invoiceable
```suggestion
                uncomplete_lines = order.order_line.filtered(
```